projects/jupyter-server-ydoc/jupyter_server_ydoc/loaders.py

```diff
@@ -204,15 +215,26 @@ async def maybe_notify(self) -> None:
         Notifies subscribed rooms about out-of-band file changes.
         """
         do_notify = False
+        filepath_change = False
         async with self._lock:
+            path = self.path
```

Comment:
Any reason to have this new `path` variable?

Reply:
To avoid having to call `self.path` several times, this function queries the database and checks certain information in the file system.

Reply:
Thanks!